packages/core/mx/contracts/staking/src/lib.rs

```diff
@@ -0,0 +1,336 @@
+        self.set_lock_period(lock_period);
+    }
+
+    #[only_owner]
+    #[endpoint(setMinimumStake)]
+    fn set_minimum_stake(&self, minimum_stake: BigUint) {
+        require!(minimum_stake > 0, "Minimum stake must be greater than 0");
+        self.minimum_stake().set(&minimum_stake);
+    }
+
+    #[only_owner]
+    #[endpoint(setLockPeriod)]
+    fn set_lock_period(&self, lock_period: u64) {
+        require!(lock_period > 0, "Must be a positive number");
+        self.lock_period().set(&lock_period);
+    }
+
+    #[only_owner]
+    #[endpoint(setRewardsPool)]
+    fn set_rewards_pool_address(&self, rewards_pool_address: ManagedAddress) {
+        self.rewards_pool_contract_address().set(&rewards_pool_address);
+    }
+
+    #[payable("*")]
+    #[endpoint(stake)]
+    fn stake(&self) {
+        let payment = self.call_value().single_esdt();
+        require!(payment.token_identifier == self.staking_token().get(), "Invalid token");
+        require!(payment.token_nonce == 0, "Invalid token nonce");
+
+        let caller = self.blockchain().get_caller();
+        let mut staker = self.get_or_create_staker(&caller);
+
+        let total_stake = staker.tokens_secure_stake() + &payment.amount;
+        require!(total_stake >= self.minimum_stake().get(), "Total stake is below the minimum threshold");
+
+        staker.deposit(&payment.amount);
+        self.stakes(&caller).set(&staker);
+
+        let stakers = self.stakers().get();
+        if !stakers.contains(&caller) {
+            self.stakers().update(|stakers| stakers.push(caller.clone()));
+        }
+        self.stake_deposited_event(&caller, &payment.amount);
+    }
+
+    #[endpoint(unstake)]
+    fn unstake(&self, tokens: BigUint) {
+        let caller = self.blockchain().get_caller();
+        require!(!self.stakes(&caller).is_empty(), "Caller is not a staker");
+
+        let mut staker = self.stakes(&caller).get();
+        require!(staker.token_staked > 0, "Must have tokens staked");
+        require!(tokens > 0, "Unstake amount must be greater than 0");
+        require!(staker.tokens_available() >= tokens, "Insufficient amount to unstake");
+
+        let new_stake = staker.tokens_secure_stake() - &tokens;
+        require!(new_stake == 0 || new_stake >= self.minimum_stake().get(), "Total stake is below the minimum threshold");
+
+        let block_number = self.blockchain().get_block_nonce();
+        let tokens_to_withdraw = staker.tokens_withdrawable(block_number);
+        if tokens_to_withdraw > 0 {
+            self.withdraw(&mut staker, &caller);
+        }
+
+        staker.lock_tokens(&tokens, self.lock_period().get(), block_number);
+        self.stakes(&caller).set(&staker);
+
+        self.stake_locked_event(&caller, &staker.tokens_locked_until, &tokens);
+    }
+
+    #[endpoint(withdraw)]
+    fn withdraw_endpoint(&self) {
+        let caller = self.blockchain().get_caller();
+        require!(!self.stakes(&caller).is_empty(), "Caller is not a staker");
```

Comment:
Save the mapper in a variable and then check if it is empty. Then you can just call the `get()` function on the mapper variable. This way you read from the storage only once.